### ISSUE TITLE
Hessian mandatory ?

### ISSUE THREAD

**Issue by edljk** [JuliaOpt/KNITRO.jl#42]
Is it possible to call KNITRO from Julia/MathPorgBase when the Hessian feature is not available ?

**Comment by mlubin** [comment]
It's not currently implemented, but you can use the logic in Ipopt as an example: https://github.com/JuliaOpt/Ipopt.jl/blob/13e38c512469a9a68dd9892df210af23cada1e48/src/IpoptSolverInterface.jl#L32

**Comment by edljk** [comment]
@mlubin Thank you for the pointer. The logic was rather simple, I managed to adapt KnitroSolverInterface.jl:

```julia
    features = features_available(d)
    has_hessian = (:Hess in features)
    if has_hessian
        initialize(d, [:Grad, :Jac, :Hess])
        Ihess, Jhess = hesslag_structure(d)
    else
        initialize(d, [:Grad, :Jac])
        Ihess = Int[]
        Jhess = Int[]
    end
    ...
    defined_hessopt = false; hessopt_value = 0
    for (param,value) in m.options
                param = string(param)
        if param == "KTR_PARAM_HESSOPT"
            defined_hessopt = true; hessopt_value = value
        end
    ...
    # check and define default hessian option 
    if (!has_hessian && !defined_hessopt) || (!has_hessian && !in(hessopt_value,2:6)) 
        setOption(m.inner,paramName2Indx["KTR_PARAM_HESSOPT"],6)
    end
```
Should I push the modification directly from my clone directory ?

**Comment by edljk** [comment]
nohessian test added and seems to be ok

```julia
nlptest_nohessian(KnitroSolver())
...
Test Passed
Expression: stat == :Optimal
Evaluated: Optimal == Optimal
```

**Comment by mlubin** [comment]
Yes, please open a PR. Thanks!

**Comment by mlubin** [comment]
Closed by #43
